packages/itwinui-react/src/core/Toast/Toaster.tsx

```diff
@@ -38,140 +35,141 @@ export type ToastOptions = Omit<
   'category' | 'isVisible' | 'id' | 'content'
 >;
 
-export default class Toaster {
-  private toasts: ToastProps[] = [];
-  private lastId = 0;
-  private settings: ToasterSettings = {
-    order: 'descending',
-    placement: 'top',
-  };
-  private toastsRef = React.createRef<ToastWrapperHandle>();
-  private isInitialized = false;
-
-  // Create container on demand.
-  // Cannot do it in constructor, because SSG/SSR apps would fail.
-  private asyncInit = async () => {
-    if (this.isInitialized) {
-      return;
-    }
-
-    const container = getContainer(TOASTS_CONTAINER_ID) ?? getDocument()?.body;
-    if (!container) {
-      return;
-    }
-    this.isInitialized = true;
-
-    const toastWrapper = <ToastWrapper ref={this.toastsRef} />;
-
-    const _ReactDOM = ReactDOM as any; // eslint-disable-line @typescript-eslint/no-explicit-any
-
-    // v18 mode
-    if (_ReactDOM.createRoot) {
-      const _ReactDOMInternals =
-        _ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
-
-      // suppress warning about importing createRoot from react-dom/client
-      if (_ReactDOMInternals) {
-        _ReactDOMInternals.usingClientEntryPoint = true;
-      }
-
-      const root = _ReactDOM.createRoot(container);
-      root.render(toastWrapper);
-      // revert suppression, not to influence users app
-      if (_ReactDOMInternals) {
-        _ReactDOMInternals.usingClientEntryPoint = false;
-      }
-    } else {
-      // v17 and before
-      ReactDOM.render(toastWrapper, container);
-    }
+// ----------------------------------------------------------------------------
+
+export const useToaster = () => {
+  const dispatch = useSafeContext(ToasterDispatchContext);
+
+  const showToast = React.useCallback(
+    (category: ToastCategory) =>
+      (content: React.ReactNode, options?: ToastOptions) => {
+        const id = nextId();
+
+        dispatch({
+          type: 'add',
+          toast: { ...options, id, content, category },
+        });
+
+        return { close: () => dispatch({ type: 'remove', id }) };
+      },
+    [dispatch],
+  );
+
+  return {
+    positive: showToast('positive'),
+    informational: showToast('informational'),
+    negative: showToast('negative'),
+    warning: showToast('warning'),
+    closeAll: () => {
+      dispatch({ type: 'close-all' });
+    },
+    setSettings: (settings: Partial<ToasterSettings>) => {
+      dispatch({ type: 'settings', settings });
+    },
   };
+};
 
-  /**
-   * Set global Toaster settings for toasts order and placement.
-   * Settings will be applied to new toasts on the page.
-   */
-  public setSettings(newSettings: ToasterSettings) {
-    newSettings.placement ??= this.settings.placement;
-    newSettings.order ??= newSettings.placement?.startsWith('bottom')
```

Comment:
Was this functionality removed on purpose?

Reply:
hmm, not intentionally. But I'm thinking about it - previously we allowed `order` to be optional, whereas now it gets a default value. We could introduce a third option `auto` and make that the default?